packages/flutter/test/painting/text_painter_test.dart

```diff
@@ -145,17 +145,79 @@ void main() {
       caretOffset = painter.getOffsetForCaret(ui.TextPosition(offset: text.length), ui.Rect.zero);
       expect(caretOffset.dx, painter.width);
 
-      // Test with trailing full-width space
-      const String textWithFullWidthSpace = 'A\u{3000}';
-      checkCaretOffsetsLtr(textWithFullWidthSpace);
-      painter.text = const TextSpan(text: textWithFullWidthSpace);
+      text = 'A';
+      painter.text = TextSpan(text: text);
       painter.layout();
-      caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 0), ui.Rect.zero);
-      expect(caretOffset.dx, 0);
-      caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 1), ui.Rect.zero);
-      expect(caretOffset.dx, painter.width / 2);
-      caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: textWithFullWidthSpace.length), ui.Rect.zero);
-      expect(caretOffset.dx, painter.width);
+      final double textAWidth = painter.width;
+      final double textAHeight = painter.height;
+      void checkUnicodeZsCategory(String character, int expectLines) {
+        text = 'A$character';
+        checkCaretOffsetsLtr(text);
+        painter.text = TextSpan(text: text);
+        painter.layout();
+        caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 0), ui.Rect.zero);
+        expect(caretOffset.dx, 0);
+        caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 1), ui.Rect.zero);
+        expect(caretOffset.dx, textAWidth);
+        caretOffset = painter.getOffsetForCaret(ui.TextPosition(offset: text.length), ui.Rect.zero);
+        expect(caretOffset.dx, painter.width);
+
+        painter.layout(maxWidth: textAWidth);
+        expect(painter.width, textAWidth);
```

Comment:
`painter.width` is not the line width. I think this should be:
```dart
final lines = painter.computeLineMetrics();
expect(lines.length, 1);
expect(lines.width, 14.0);
```

Reply:
I added test case at [86adcc1](https://github.com/flutter/flutter/pull/152215/commits/86adcc1f8e9652df3f7a054c1258529f5d6fd15e).